code/clue.js

```diff
@@ -1,59 +1,548 @@
+  const randomIndex = Math.floor(Math.random() * randomArray.length);
+
+  // Returning the randomIndex ---> [] from the randomArray
+  return randomArray[randomIndex];
+};
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+// *** Clicking "KILLER CARD" ***
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  // flips card
+  document.getElementById("killerCard").classList.add("flip");
+
+  // time delay starts
+  setTimeout(() => {
+    // random killer selector
+    mystery.killer = randomSelector(suspects);
+
+    // *** changing card info ***
+
+    // card background color
+    document.getElementById("killerCard").style.background = mystery.killer.color;
+    // heading background color
+    document.getElementById("cardHeadingKiller").style.background = mystery.killer.heading;
+    // killer name
+    document.getElementById("killerName").innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+    // killer occupation
+    document.getElementById("killerOccupation").innerHTML = `${mystery.killer.occupation}`;
+    //killer image
+    document.getElementById("killerImage").src = `${mystery.killer.image}`;
+
+    // time delay millisecond
+  }, 300);
+};
+
+document.getElementById("killerCard").addEventListener("click", pickKiller);
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+// *** Clicking "WEAPON CARD" ***
+
+const pickWeapon = () => {
+  // validating killer-card
+  if (mystery.killer) {
+    // flips card
+    document.getElementById("weaponCard").classList.add("flip");
+
+    // finding the weapon object with the killers favourite weapon id
+    const favouriteWeapon = weapons.find(
+      (weapon) => weapon.name === mystery.killer.favouriteWeaponId
+    );
+
+    // time delay starts
+    setTimeout(() => {
+      // "most likely" random weapon selector
+      mystery.weapon = mostLikelyRandomSelector(weapons, favouriteWeapon);
+
+      // *** changing card info ***
+
+      // card background color
+      document.getElementById("weaponCard").style.background = mystery.weapon.color;
+      // heading background color
+      document.getElementById("cardHeadingWeapon").style.background = mystery.weapon.heading;
+      // weapon name
+      document.getElementById("weaponName").innerHTML = `${mystery.weapon.name}`;
+      // weapon weight
+      document.getElementById("weaponWeight").innerHTML = `${mystery.weapon.weight} of weight`;
+      // weapon image
+      document.getElementById("weaponImage").src = `${mystery.weapon.image}`;
+
+      // time delay millisecond
+    }, 300);
+  } /* else {
+    document.getElementById("mystery").innerHTML = "Select a killer card!";
+  } */
+};
```

Comment:
With such similar code in several places, it would be interesting to consider how we could use the same code for the room, killer and weapon.